src/config.rs

```diff
@@ -344,13 +348,22 @@ impl CommandNameAndArgs {
             CommandNameAndArgs::Vec(NonEmptyCommandArgsVec(a)) => {
                 (Cow::Borrowed(&a[0]), Cow::Borrowed(&a[1..]))
             }
+            CommandNameAndArgs::Structured {
```

Comment:
If I understand correctly, this makes the configuration inside `[merge-tools.TOOL]` even more different from the configuration inside `ui.diff-editor`. We'll probably need to rework the former.

Reply:
Yeah, we could add `edit-env`/`merge-env`/etc. for completeness. So long as the structured syntax is undocumented, I think it's okay to leave merge-tools not supporting env table.

Reply:
I'm wondering if we could redesign it in a way so that there are not two separate codepaths. I haven't thought it through, though. (Obviously, this is entirely unrelated to this PR).

Reply:
I suppose we could change the config to be more like this:

```
ui.pager = {env = {...}, command = "less", args = ["-FRX"]}
merge-tools.vimdiff.merge-args = { env = {...}, command = "vim", args = ["-f", "-d", "$output", ...]}
```

`command` in there could default to the tool name. It would be kind of weird that you could use different command names for `merge-args` and `diff-args`.